Sources/SmithyCodegenCLI/SmithyCodegenCLI.swift

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import ArgumentParser
+import Foundation
+import SmithyCodegenCore
+
```

Comment:
This is a very simple code generator that:
- Reads the Smithy JSON model file into memory.
- Generates a Swift source file with a public variable `modelCount: Int` that contains the number of bytes in the model file.
- Adds a simple doc comment to the variable.

While this example codegen is trivial and not useful in itself, it does prove that:
- A Swift build plugin code generator can read from our model files elsewhere in the project.
- It can generate a Swift file that is added to the target.
- Doc comments in generated code get picked up with the rest of the target's doc comments.

Follow-on development would code-generate more substantial content from the model.